connexion/decorators/uri_parsing.py

```diff
@@ -201,6 +201,9 @@ def _make_deep_object(self, k, v):
             if k == root_key:
                 return k, v, False
 
+        if not self._is_deep_object_style_param(root_key):
+            return k, v, False
+
```

Comment:
This is the first place where we know the root parameter name and have the schema information available. We already have a lot of special casing here, so that follow-up code can assume everything is fine. Let me know if you prefer to move that checker somewhere else.